### PR TITLE
P4-2493 initial commit for start of work for mapping friendly location names.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
There is no way via the app at the moment to get to this page.  You can for now (once logged in) go via a get request to:

 **/map-location/<PUT_AGENCY_ID_HERE>**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MapFriendlyLocationService.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import uk.gov.justice.digital.hmpps.pecs.jpc.config.TimeSource
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.Location
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationType
+
+@Service
+@Transactional
+class MapFriendlyLocationService(private val locationRepository: LocationRepository, private val timeSource: TimeSource) {
+  fun mapFriendlyLocation(agencyId: String, friendlyLocationName: String, locationType: LocationType) {
+    locationRepository.save(Location(locationType, agencyId.toUpperCase().trim(), friendlyLocationName.toUpperCase().trim(), timeSource.dateTime()))
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/MapFriendlyLocationService.kt:14]: So no checks for duplicates at this point so will just fail with a low level error.

File: src/main/resources/templates/fragments/header.html [hunk 2]
```diff
@@ -20,8 +20,8 @@
                 </a>
             </div>
             <div sec:authorize="isAuthenticated()" class="govuk-header__content tr mt1">
-                <span class="govuk-!-font-weight-bold pr4" sec:authentication="name"></span>
                 <form id="logoutForm" method="POST" th:action="@{/logout}">
+                    <span class="govuk-!-font-weight-bold pr4" sec:authentication="name"></span>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/fragments/header.html:24]: Just sorting out the positioning of the....

File: src/main/resources/templates/map-location.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/map-location.html:1]: This page will need some love @danielepolencic .  Not my speciality.

File: src/main/resources/templates/map-location.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                Map location
+            </label>
+            </h1>
+            <div id="map-location-name-hint" class="govuk-hint">
+                Please add a Schedule 34 location name for this NOMIS agency ID.
+            </div>
+            <div id="agency-id-hint" class="govuk-hint">
+                <span class="db govuk-!-font-weight-bold">Pick up location</span>
+                <p th:text="*{agencyId}"></p>
+                <input type="hidden" th:field="*{agencyId}"/>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/map-location.html:27]: Should/could this be a session attribute instead?

Comment (jcavell) [reply src/main/resources/templates/map-location.html:27]: No i think hidden is good

**Review by jcavell — APPROVED** [review]

**Pull request merged**